### ISSUE TITLE
TerminateProcess used to exit in response to SIGINT

### ISSUE THREAD

**Issue by stephentoub** [dotnet/coreclr#1096]
In SEHHandleControlEvent, if there aren't any ctrl-C handlers or none of them handle the event, the process exits.  However, it does so using TerminateProcess, which results in the process generating a core dump and printing out "Aborted (core dump)" to the terminal.  This is not the proper way a program on Unix should exit in response to SIGINT / ctrl-C.

See "how to be a proper program" at http://www.cons.org/cracauer/sigint.html.

**Comment by kangaroo** [comment]
The issue we call TerminateProcess and end up here:

```
        if (bTerminateUnconditionally)
        {
            // abort() has the semantics that 
            // (1) it doesn't run atexit handlers
            // (2) can invoke CrashReporter or produce a coredump,
            // which is appropriate for TerminateProcess calls
            
            // If this turns out to be inappropriate for some case, where we
            // call TerminateProcess vs. ExitProcess, then there needs to be
            // a CLR wrapper for TerminateProcess and some exposure for PAL_abort()
            // to selectively use that in all but those cases.
            
            abort();
        }
```

For Ctrl-C we probably just want ExitProcess, rather than TerminateProcess?

**Comment by janvorli** [comment]
Fixed as #1105
